### PR TITLE
Feature/market attribute

### PR DESCRIPTION
**Pull request by pvlasak**
Ahoj, sdilim prvni verzi upravy tykajici se atributu market v run manager record. Zitra v meetingu muzeme probrat.

### CODE REVIEW

File: v2realbot/common/model.py [hunk 1]
```diff
@@ -159,6 +159,7 @@ class RunManagerRecord(BaseModel):
     mode: Mode
     note: Optional[str] = None
     ilog_save: bool = False
+    market: Optional[Market] = None
```
Comment (drew2323) [inline v2realbot/common/model.py:162]: def. value US

File: v2realbot/common/transform.py [hunk 2]
```diff
@@ -17,6 +16,7 @@ def row_to_runmanager(row: dict) -> RunManagerRecord:
         account=row['account'],
         note=row['note'],
         ilog_save=bool(row['ilog_save']),
+        market=str(row['market']) if row['market'] is not None else None,
```
Comment (drew2323) [inline v2realbot/common/transform.py:19]: nemusi byt str()

File: v2realbot/controller/run_manager.py [hunk 3]
```diff
@@ -172,14 +172,14 @@ def add_run_manager_record(new_record: RunManagerRecord):
...
+                                    market, bt_from, bt_to, weekdays_filter, batch_id,
                                     start_time, stop_time, status, last_processed,
                                     history, valid_from, valid_to, testlist_id)
-            VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) 
+            VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?,?) 
         """
         values = [
             new_record.moddus, str(new_record.id), str(new_record.strat_id), new_record.symbol, new_record.account, new_record.mode, new_record.note,
-            int(new_record.ilog_save),
+            int(new_record.ilog_save), 'US',
```
Comment (drew2323) [inline v2realbot/controller/run_manager.py:182]: new_record.market

File: v2realbot/controller/run_manager.py [hunk 4]
```diff
@@ -244,6 +244,8 @@ def update_run_manager_record(record_id, updated_record: RunManagerRecord):
             update_query += f"{key} = ?, "
             if key == "ilog_save":
                 value = int(value)
+            elif key == "market":
+                value = 'US'
```
Comment (drew2323) [inline v2realbot/controller/run_manager.py:248]: value = str(value)

File: v2realbot/controller/run_manager.py [hunk 5]
```diff
@@ -376,6 +378,23 @@ def fetch_scheduled_candidates_for_start_and_stop(market_datetime_now, market) -
         conn.row_factory = None
         db.pool.release_connection(conn)
 
+def fetch_all_markets_in_run_manager() -> list:
```
Comment (drew2323) [inline v2realbot/controller/run_manager.py:381]: asi pryč

File: v2realbot/scheduler/ap_scheduler.py [hunk 6]
```diff
@@ -113,18 +113,20 @@ def initialize_jobs(run_manager_records: RunManagerRecord = None):
                                     start_date=record.valid_from, end_date=record.valid_to, timezone=zoneNY)
 
             # Schedule new jobs with the 'scheduler_' prefix
-            scheduler.add_job(start_runman_record, start_trigger, id=f"scheduler_start_{record.id}", args=[record.id])
-            scheduler.add_job(stop_runman_record, stop_trigger, id=f"scheduler_stop_{record.id}", args=[record.id])
+            market_value = record.market.value if record.market else None
```
Comment (drew2323) [inline v2realbot/scheduler/ap_scheduler.py:116]: do jobu jako argument pauze record.id , market se pak dotahne dle id v _start_runman_record

File: v2realbot/scheduler/ap_scheduler.py [hunk 7]
```diff
@@ -113,18 +113,20 @@ def initialize_jobs(run_manager_records: RunManagerRecord = None):
...
+    # f"scheduler_testinterval")
     scheduled_jobs = scheduler.get_jobs()
     print(f"APS jobs refreshed ({len(scheduled_jobs)})")
     current_jobs_dict = format_apscheduler_jobs(scheduled_jobs)
     richprint(current_jobs_dict)
     return 0, current_jobs_dict
 
 #zastresovaci funkce resici error handling a printing
-def start_runman_record(id: UUID, market = "US", debug_date = None):
+def start_runman_record(id: UUID, market, debug_date = None):
```
Comment (drew2323) [inline v2realbot/scheduler/ap_scheduler.py:129]: market pryc

File: v2realbot/scheduler/ap_scheduler.py [hunk 8]
```diff
@@ -113,18 +113,20 @@ def initialize_jobs(run_manager_records: RunManagerRecord = None):
...
     record = None
     res, record, msg = _start_runman_record(id=id, market=market, debug_date=debug_date)
```
Comment (drew2323) [inline v2realbot/scheduler/ap_scheduler.py:131]: market pryc

File: v2realbot/scheduler/ap_scheduler.py [hunk 9]
```diff
@@ -165,7 +167,7 @@ def update_runman_record(record: RunManagerRecord):
         err_msg= f"STOP: Error updating {record.id} errir {set} with values {record}"
         return -2, err_msg#toto stopne zpracovani dalsich zaznamu pri chybe, zvazit continue
 
-def stop_runman_record(id: UUID, market = "US", debug_date = None):
+def stop_runman_record(id: UUID, market, debug_date = None):
```
Comment (drew2323) [inline v2realbot/scheduler/ap_scheduler.py:170]: market pryc

File: v2realbot/scheduler/ap_scheduler.py [hunk 10]
```diff
@@ -165,7 +167,7 @@ def update_runman_record(record: RunManagerRecord):
...
     res, record, msg = _stop_runman_record(id=id, market=market, debug_date=debug_date)
```
Comment (drew2323) [inline v2realbot/scheduler/ap_scheduler.py:171]: market pryc

File: v2realbot/scheduler/ap_scheduler.py [hunk 11]
```diff
@@ -207,11 +209,13 @@ def _start_runman_record(id: UUID, market = "US", debug_date = None):
     
     record = result
 
-    if market is not None and market == "US":
+    if market == "US" or market is None:
```
Comment (drew2323) [inline v2realbot/scheduler/ap_scheduler.py:212]: record.market

File: v2realbot/scheduler/ap_scheduler.py [hunk 12]
```diff
@@ -207,11 +209,13 @@ def _start_runman_record(id: UUID, market = "US", debug_date = None):
...
         res, sada = sch.get_todays_market_times(market=market, debug_date=debug_date)
         if res == 0:
             market_time_now, market_open_datetime, market_close_datetime = sada
             print(f"OPEN:{market_open_datetime} CLOSE:{market_close_datetime}")
+        elif res == 1:
```
Comment (drew2323) [inline v2realbot/scheduler/ap_scheduler.py:217]: market None rusime, typ marketu bude vzdy dany enumeraci s danym defaultem

File: v2realbot/scheduler/scheduler.py [hunk 13]
```diff
@@ -28,17 +37,15 @@ def get_todays_market_times(market = "US", debug_date = None):
...
             #zatim podpora pouze main session
-
             #pouze main session
-            market_open_datetime = zoneNY.localize(cal_dates[0].open)
-            market_close_datetime = zoneNY.localize(cal_dates[0].close)
-            return 0, (nowNY, market_open_datetime, market_close_datetime)
+                market_open_datetime = zoneNY.localize(calendar_dates[0].open)
+                market_close_datetime = zoneNY.localize(calendar_dates[0].close)
+                return 0, (nowNY, market_open_datetime, market_close_datetime)
+        elif market is None:
```
Comment (drew2323) [inline v2realbot/scheduler/scheduler.py:47]: market None rusime, vytvarime vidlicku pro CRYPTO, ktera bude vracet nowNY, a open cas a close cas

**Review by drew2323 — CHANGES_REQUESTED** [review]
see comments

File: v2realbot/common/model.py [hunk 14]
```diff
@@ -159,7 +159,7 @@ class RunManagerRecord(BaseModel):
     mode: Mode
     note: Optional[str] = None
     ilog_save: bool = False
-    market: Optional[Market] = None
+    market: Optional[Market] = 'US'
```
Comment (drew2323) [inline v2realbot/common/model.py:162]: Market.US

File: v2realbot/controller/run_manager.py [hunk 15]
```diff
@@ -245,7 +245,10 @@ def update_run_manager_record(record_id, updated_record: RunManagerRecord):
             if key == "ilog_save":
                 value = int(value)
             elif key == "market":
```
Comment (drew2323) [inline v2realbot/controller/run_manager.py:247]: zkusit pryc celou sekci

File: v2realbot/scheduler/ap_scheduler.py [hunk 16]
```diff
@@ -114,8 +114,8 @@ def initialize_jobs(run_manager_records: RunManagerRecord = None):
 
             # Schedule new jobs with the 'scheduler_' prefix
             market_value = record.market.value if record.market else None
```
Comment (drew2323) [inline v2realbot/scheduler/ap_scheduler.py:116]: pryc

File: v2realbot/scheduler/scheduler.py [hunk 17]
```diff
@@ -29,7 +29,7 @@ def is_market_day(date):
 
 def get_todays_market_times(market, debug_date = None):
     try:
-        if market == "US":
+        if market == "US" or market == "CRYPTO":
```
Comment (drew2323) [inline v2realbot/scheduler/scheduler.py:32]: 1) pouzit Market.US a Market.CRYPTO
2) pro market US nechat jak je (tzn. dotahnout z is_market_day fce)
3) vytvorit vetev Market.CRYPTO - která vrátí
- now_market_datetime - nowNY(dle variable CRYPTO_TIMEZONE = zoneNY) datetime.now().astimezone(zoneNY)
- market_open_datetime - začátek now_market_datetime (datum + 0:00:00)
- matket_close_datetime - konec now_market_datetime  (datum + 23:59:59)

**Comment by pvlasak** [comment]
Instructions:
---------------------
1. connect to SQLite DB and add a column named market to run_manager table: 

import sqlite3
from v2realbot.common.db import sqlite_db_file

conn = sqlite3.connect(sqlite_db_file)
cursor = conn.cursor()
cursor.execute("""ALTER TABLE run_manager ADD COLUMN market TEXT;""")
cursor.execute("""UPDATE run_manager SET market='US';""")
conn.commit()
conn.close()

3. Use application as usual. Any new run manager record is set by default for US stock market
4. You can also use sequence of SQL commands in `v2trading_create_db.sql` to create a new SQLite database from scratch

**Review by drew2323 — APPROVED** [review]

**Pull request merged**